Scripts/muemu.py

```diff
@@ -0,0 +1,257 @@
+    # qemu_args += ["-device", "usb-tablet"]
+
+    # Network
+    qemu_args += ["-nic", "model=e1000"]
+
+    # TPM
+    if args.tpm:
+        qemu_args += ["-chardev",
+                      f"socket,id=chrtpm,path={args.firmwaredir}/tpm/swtpm-sock"]
+        qemu_args += ["-tpmdev", "emulator,id=tpm0,chardev=chrtpm"]
+        qemu_args += ["-device", "tpm-tis,tpmdev=tpm0"]
+
+    # Display
+    if args.vnc != None:
+        qemu_args += ["-display", f"vnc={args.vnc}"]
+
+    # Debug & Serial ports
+    if args.gdbport != None:
+        qemu_args += ["-gdb", f"tcp::{args.gdbport}"]
+
+    # Launch QEMU
+    run_qemu(qemu_args)
+
+
+def build_args_x64(qemu_args: List[str]):
+    smm_value = "off" if args.accel == "whpx" else "on"
+    qemu_args += [f"{args.qemudir}qemu-system-x86_64"]
+    qemu_args += ["-cpu", "qemu64,+rdrand,umip,+smep,+popcnt"]
+    qemu_args += ["-global", "ICH9-LPC.disable_s3=1"]
+    qemu_args += ["-machine", f"q35,smm={smm_value},accel={args.accel}"]
+    qemu_args += ["-debugcon", "stdio"]  # file:uefi-x64.log
+    qemu_args += ["-global", "isa-debugcon.iobase=0x402"]
+    qemu_args += ["-vga", "cirrus"]
+
+    # Flash storage
+    if smm_value == "on":
+        code_fd = f"{args.firmwaredir}/x64/QemuQ35/VisualStudio-x64/QEMUQ35_CODE.fd"
+        data_fd = f"{args.firmwaredir}/x64/QemuQ35/VisualStudio-x64/QEMUQ35_VARS.fd"
+        qemu_args += ["-global",
+                      "driver=cfi.pflash01,property=secure,value=on"]
+    else:
+        print("Switching to no-SMM firmware for WHPX.")
+        code_fd = f"{args.firmwaredir}/x64/QemuQ35.NoSmm/VisualStudio-NoSmm-x64/QEMUQ35_CODE.fd"
+        data_fd = f"{args.firmwaredir}/x64/QemuQ35.NoSmm/VisualStudio-NoSmm-x64/QEMUQ35_VARS.fd"
+
+    qemu_args += ["-drive",
+                  f"if=pflash,format=raw,unit=0,file={code_fd},readonly=on"]
+    qemu_args += ["-drive", f"if=pflash,format=raw,unit=1,file={data_fd}"]
+
+    if args.cores > 4 and not args.force:
+        print("Only 4 core currently supported for ARM64, setting cores to 4.")
+        args.cores = 4
+
+
+def build_args_arm64(qemu_args: List[str]):
+    qemu_args += [f"{args.qemudir}qemu-system-aarch64"]
+    qemu_args += ["-machine", f"sbsa-ref,accel={args.accel}"]
+    qemu_args += ["-cpu", "max"]
+    qemu_args += ["-serial", "stdio"]  # file:uefi-arm64.log
+
+    # Flash storage
+    sec_fd = f"{args.firmwaredir}/aarch64/QemuSbsa/GCC-AARCH64/SECURE_FLASH0.fd"
+    efi_fd = f"{args.firmwaredir}/aarch64/QemuSbsa/GCC-AARCH64/QEMU_EFI.fd"
+    qemu_args += ["-global", "driver=cfi.pflash01,property=secure,value=on"]
+    qemu_args += ["-drive", f"if=pflash,format=raw,unit=0,file={sec_fd}"]
+    qemu_args += ["-drive",
+                  f"if=pflash,format=raw,unit=1,file={efi_fd},readonly=on"]
+
+    if args.cores != 4 and not args.force:
+        print("Arm64 image must use 4 cores! Setting to 4.")
+        args.cores = 4
+
+
+def run_qemu(qemu_args: List[str]):
+    if args.tpm:

```

Comment:
Set `swtpm_proc` to `None` here to prevent an "unbound error" on when the value is compared to `None` later in the function and `args.tpm` is `False`.

```suggestion
def run_qemu(qemu_args: List[str]):
    swtpm_proc = None

    if args.tpm:
```